### PR TITLE
Added reverse sorting

### PR DESCRIPTION
**Pull request by Revadike**
image

### CODE REVIEW

File: js/content/community.js [hunk 1]
```diff
@@ -1523,6 +1523,8 @@ let StatsPageClass = (function(){
             let node = item[1];
             personal.insertAdjacentElement("beforeend", node);
         }
+
+        SyncedStorage.set("sortachievementsby", key);
```
Comment (MxtOUT) [inline js/content/community.js]: Stuff like that is normally saved in `LocalStorage`

File: js/content/community.js [hunk 2]
```diff
@@ -2863,7 +2873,17 @@ let FriendsThatPlayPageClass = (function(){
         defaultNode.insertAdjacentElement("afterend", sorted);
         HTML.afterEnd(defaultNode, "<div style='clear: both'></div>");
 
+        let active = "default";
```
Comment (MxtOUT) [inline js/content/community.js]: Could you save this information with CSS classes?

Comment (Revadike) [reply js/content/community.js]: Does that matter?

File: js/content/community.js (hunk 1 above)
Comment (Revadike) [reply js/content/community.js]: All other existing sorting functions also used `SyncedStorage`, I was just following their example.

File: js/content/community.js (hunk 2 above)
Comment (MxtOUT) [reply js/content/community.js]: It's more idiomatic

File: js/content/community.js (hunk 1 above)
Comment (MxtOUT) [reply js/content/community.js]: I just had this example of the HD playback option in my mind.
In the end I think it doesn't really matter, `SyncedStorage` is more persistent, but `LocalStorage` is more suitable for such information imo.
We should only use one though

File: js/content/community.js (hunk 2 above)
Comment (MxtOUT) [reply js/content/community.js]: i.e. following certain conventions

**Comment by Revadike** [comment]
> I feel like you need to show whether reverse sorting is currently active

Wouldn't it make more sense to show what type of sorting is active, whether it is reversed or not?

File: js/content/community.js (hunk 2 above)
Comment (Revadike) [reply js/content/community.js]: Well, in other sorting pages, this state is maintained in LocalStorage instead. However, for some reason here there was no sorting state remembering, thus I didn't add it either. Frankly, the entire friendsthatplay page sorting is pretty broken. It doesn't properly sort everything and also breaks sometimes if it cannot find a play time.

**Comment by MxtOUT** [comment]
Why not both?
image

**Comment by Revadike** [comment]
> Why not both?
> image

Because it conflicts with the existing logic and it would be rather labor intensive to change all that.

**Comment by MxtOUT** [comment]
You'd need just two images (one with an arrow pointing down, the other up) and apply some CSS rules

**Comment by Revadike** [comment]
You could even use 1 image and rotate it. Or no image and use ASCII.

**Comment by MxtOUT** [comment]
9ac6043151b934678292ef2dced1aa38374f0531 integrates the reverse sorting feature with `SearchBox`

**Comment by Revadike** [comment]
What about group searching?

**Comment by Revadike** [comment]
Also, you forgot to add it to the friendsthatplay page. Granted, the sorting there is currently broken, so you'd need to fix that too. Should I make a separate issue for it?

**Comment by MxtOUT** [comment]
Not needed, I'll mark this thread as unread, thanks

**Comment by Revadike** [comment]
Found another bug:
image

image

This happens if you load a friends page of somebody other than yourself.

**Comment by MxtOUT** [comment]
Thanks, fixed with b998152da040283ed9656d46bf8322dbe9dbc640

**Pull request closed without merging**